libs/ledger-live-common/src/hw/actions/ftsLoadImage.ts

```diff
@@ -0,0 +1,352 @@
+} from "@ledgerhq/errors";
+import { getDeviceModel } from "@ledgerhq/devices";
+
+type State = {
+  isLoading: boolean;
+  requestQuitApp: boolean;
+  unresponsive: boolean;
+  imageLoadRequested?: boolean;
+  loadingImage?: boolean;
+  imageLoaded?: boolean;
+  imageCommitRequested?: boolean;
+  imageHash?: string;
+  device: Device | null | undefined;
+  deviceInfo: DeviceInfo | null | undefined;
+  error: Error | null | undefined;
+  progress?: number;
+};
+
+type LoadImageAction = Action<string, State, { imageHash?: string }>;
+
+const mapResult = ({ imageHash }: State) => ({ imageHash });
+
+type Event =
+  | LoadImageEvent
+  | {
+      type: "error";
+      error: Error;
+    }
+  | {
+      type: "deviceChange";
+      device: Device | null | undefined;
+    };
+
+const getInitialState = (device?: Device | null | undefined): State => ({
+  isLoading: !!device,
+  requestQuitApp: false,
+  unresponsive: false,
+  device,
+  deviceInfo: null,
+  error: null,
+});
+
+const reducer = (state: State, e: Event): State => {
+  switch (e.type) {
+    case "unresponsiveDevice":
+      return { ...state, unresponsive: true, isLoading: false };
+
+    case "deviceChange":
+      return getInitialState(e.device);
+
+    case "error":
+      return {
+        ...getInitialState(state.device),
+        error: e.error,
+        isLoading: false,
+      };
+    case "appDetected":
+      return {
+        ...state,
+        unresponsive: false,
+        requestQuitApp: true,
+        isLoading: false,
+      };
+    case "loadImagePermissionRequested":
+      return {
+        ...state,
+        unresponsive: false,
+        imageLoadRequested: true,
+        isLoading: false,
+      };
+    case "commitImagePermissionRequested":
+      return {
+        ...state,
+        unresponsive: false,
+        imageCommitRequested: true,
+        loadingImage: false,
+        isLoading: false,
+      };
+    case "imageLoaded":
+      return {
+        ...state,
+        unresponsive: false,
+        imageLoadRequested: false,
+        imageCommitRequested: false,
+        isLoading: false,
+        loadingImage: false,
+        imageLoaded: true,
+      };
+    case "progress":
+      return {
+        ...state,
+        unresponsive: false,
+        imageLoadRequested: false,
+        imageCommitRequested: false,
+        isLoading: false,
+        loadingImage: true,
+        progress: e.progress,
+      };
+  }
+};
+
+const implementations = {
+  // in this paradigm, we know that deviceSubject is reflecting the device events
+  // so we just trust deviceSubject to reflect the device context (switch between apps, dashboard,...)
+  event: ({ deviceSubject, loadImage, language }) =>
+    deviceSubject.pipe(
+      debounceTime(1000),
+      switchMap((d) => loadImage(d, language))
+    ),
+  // in this paradigm, we can't observe directly the device, so we have to poll it
+  polling: ({ deviceSubject, loadImage, language }) =>
```

Comment:
we'll need to handle the new `5515` locked device here - because i guess with the new fwm, if the device is locked, you'll have an error. But let's not do this in this PR ;)